Gems/Atom/Utils/Code/Source/ImGuiGpuProfiler.cpp

```diff
@@ -2014,14 +2014,17 @@ namespace AZ
                 const RPI::ParentPass* passAsParent = pass->AsParent();
 
                 // Add new entry to the timestamp map.
-                PassEntry* entry = addPassEntry(pass, parent);
-
-                // Recur if it's a parent.
-                if (passAsParent)
+                if (pass->IsEnabled())
                 {
-                    for (const auto& childPass : passAsParent->GetChildren())
+                    PassEntry* entry = addPassEntry(pass, parent);
```

Comment:
nit: We could move `entry` inside the scope of `passAsParent` as it's not needed outside that scope and is closer to first use

Reply:
I don't think we can. 
The entry is return value of addPassEntry(). And addPassEntry needs to be called outside passAsParent.